website/src/components/Messages/MessageTableEntry.tsx

```diff
@@ -281,9 +311,11 @@ const MessageActions = ({
             <MenuItem as="a" href={`/admin/manage_user/${message.user_id}`} target="_blank" icon={<User />}>
               {t("view_user")}
             </MenuItem>
-            <MenuItem onClick={handleDelete} icon={<Trash />}>
-              {t("common:delete")}
-            </MenuItem>
+            {!message.deleted && (
+              <MenuItem onClick={handleDelete} icon={<Trash />}>
+                {t("common:delete")}
+              </MenuItem>
+            )}
```

Comment:
Now it possible to show deleted message, so I hide the delete button when message is already deleted.